mutmut/node_mutation.py

```diff
@@ -125,6 +128,62 @@ def operator_string_methods_swap(
+            if old_call in {"split", "rsplit"}:
+                # split() -> split()                                        # not len(node.args)
+                # rsplit() -> rsplit()                                      # not len(node.args)
+                # split(" ") -> split(" ")                                  # len(node.args) == 1 & maxsplit not in args
+                # rsplit(" ") -> rsplit(" ")                                # len(node.args) == 1 & maxsplit not in args
+                # split(sep=" ") -> split(sep=" ")                          # len(node.args) == 1 & maxsplit not in args
+                # rsplit(sep=" ") -> rsplit(sep=" ")                        # len(node.args) == 1 & maxsplit not in args
+                # split(maxsplit=-1) -> split(maxsplit=-1)                  # maxsplit=-1 in args
+                # rsplit(maxsplit=-1) -> rsplit(maxsplit=-1)                # maxsplit=-1 in args
+                # split(" ", maxsplit=-1) -> split(" ", maxsplit=-1)        # maxsplit=-1 in args
+                # rsplit(" ", maxsplit=-1) -> rsplit(" ", maxsplit=-1)      # maxsplit=-1 in args
+
+                # split(maxsplit=1) -> rsplit(maxsplit=1)                   # maxsplit in args and maxsplit != -1
+                # rsplit(maxsplit=1) -> split(maxsplit=1)                   # maxsplit in args and maxsplit != -1
+
+                # split(" ", 1) -> rsplit(" ", 1)                           # len(node.args) == 2 & maxsplit not in args
+                # rsplit(" ", 1) -> split(" ", 1)                           # len(node.args) == 2 & maxsplit not in args
+
+                # split(" ", maxsplit=1) -> rsplit(" ", maxsplit=1)         # maxsplit in args and maxsplit != -1
+                # rsplit(" ", maxsplit=1) -> split(" ", maxsplit=1)         # maxsplit in args and maxsplit != -1
+                key_args: set[str] = {a.keyword.value for a in node.args if a.keyword}
+
+                maxsplit_val = None
+                for a in node.args:
+                    if a.keyword and a.keyword.value == "maxsplit":
+                        if any([isinstance(child, cst.Minus) for child in a.value.children]):
+                            maxsplit_val = int(a.value.expression.value) * -1
+                            break
+                        if any([isinstance(child, cst.Plus) for child in a.value.children]):
+                            maxsplit_val = int(a.value.expression.value)
+                            break
+                        maxsplit_val = a.value.evaluated_value
+
+
+                if (
+                        not len(node.args) or
+                        (len(node.args) == 1 and "maxsplit" not in key_args) or
+                        ("maxsplit" in key_args and maxsplit_val == -1)
+                ):
+                    continue
+
+                if (
+                        (len(node.args) == 2 and "maxsplit" not in key_args) or
```

Comment:
It cannot happen that we have two args *and* `maxsplit` as a keyword arg (we cannot pass the argument both as a positional argument and keyword argument). So we could remove the ` and "maxsplit" not in key_args` part here.